ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceITBase.scala

```diff
@@ -0,0 +1,194 @@
+
+abstract class ResetServiceITBase
+    extends AsyncWordSpec
+    with AsyncTimeLimitedTests
+    with Matchers
+    with ScalaFutures
+    with AbstractSandboxFixture
+    with SuiteResourceManagementAroundEach
+    with TestCommands {
+
+  override def timeLimit: Span = scaled(30.seconds)
+
+  override protected def config: SandboxConfig =
+    super.config.copy(ledgerIdMode = LedgerIdMode.Dynamic)
+
+  protected val eventually: RetryStrategy = RetryStrategy.exponentialBackoff(10, 10.millis)
+
+  override protected def darFile: File = new File(rlocation("ledger/test-common/Test-stable.dar"))
+
+  protected def fetchLedgerId(): Future[String] =
+    LedgerIdentityServiceGrpc
+      .stub(channel)
+      .getLedgerIdentity(GetLedgerIdentityRequest())
+      .map(_.ledgerId)
+
+  // Resets and waits for a new ledger identity to be available
+  protected def reset(ledgerId: String): Future[String] =
+    for {
+      _ <- ResetServiceGrpc.stub(channel).reset(ResetRequest(ledgerId))
+      newLedgerId <- eventually { (_, _) =>
+        fetchLedgerId()
+      }
+    } yield newLedgerId
+
+  protected def timedReset(ledgerId: String): Future[(String, FiniteDuration)] = {
+    val start = System.nanoTime()
+    reset(ledgerId).map(_ -> (System.nanoTime() - start).nanos)
+  }
+
+  protected def allocateParty(hint: String): Future[String] =
+    PartyManagementServiceGrpc
+      .stub(channel)
+      .allocateParty(AllocatePartyRequest(hint))
+      .map(_.partyDetails.get.party)
+
+  protected def submitAndWait(req: SubmitAndWaitRequest): Future[Empty] =
+    CommandServiceGrpc.stub(channel).submitAndWait(req)
+
+  protected def activeContracts(ledgerId: String, f: TransactionFilter): Future[Set[CreatedEvent]] =
+    new StreamConsumer[GetActiveContractsResponse](
+      ActiveContractsServiceGrpc
+        .stub(channel)
+        .getActiveContracts(GetActiveContractsRequest(ledgerId, Some(f)), _))
+      .all()
+      .map(_.flatMap(_.activeContracts)(collection.breakOut))
+
+  protected def submitAndExpectCompletions(
+      ledgerId: String,
+      commands: Int,
+      party: String,
+  ): Future[Unit] =
+    for {
+      _ <- Future.sequence(
+        Vector.fill(commands)(
+          CommandSubmissionServiceGrpc
+            .stub(channel)
+            .submit(dummyCommands(LedgerId(ledgerId), UUID.randomUUID.toString, party))))
+      unit <- WaitForCompletionsObserver(commands)(
+        CommandCompletionServiceGrpc
+          .stub(channel)
+          .completionStream(
+            CompletionStreamRequest(
+              ledgerId = ledgerId,
+              applicationId = M.applicationId,
+              parties = Seq(party),
+              offset = Some(M.ledgerBegin)
+            ),
+            _))
+    } yield unit
+
+  "ResetService" when {
+    "state is reset" should {
+      "return a new ledger ID" in {
+        for {
+          lid1 <- fetchLedgerId()
+          lid2 <- reset(lid1)
+          throwable <- reset(lid1).failed
```

Comment:
This doesn't strictly match directly with the test description, although transitively it's still testing that the ledger id had changed.

Reply:
Yeah, I tried not to change the tests (just moved them from `ResetServiceIT` to `ResetServiceITBase`), but you're right, this one is confusing. How would you change it?

Reply:
Hm, actually it's fine as it is. The test compares `lid1` and `lid2`, and sending any request with the old ledgerId should be rejected ... might as well try to reset again.